pennylane/ops/op_math/exp.py

```diff
@@ -279,9 +279,22 @@ def _recursive_decomposition(self, base: Operator, coeff: complex):
             ops = base.ops if isinstance(base, Hamiltonian) else base.operands
             return self._trotter_decomposition(ops, coeffs)
 
-        for op_class in has_unitary_generator_types:
-            # Check if the exponentiated operator is a generator of another operator
-            if op_class.num_wires in {base.num_wires, AnyWires} and op_class is not qml.PauliRot:
+        # Store operator classes with generators
+        has_generator_types = []
+        has_generator_types_anywires = []
+        for op_name in qml.ops.qubit.__all__:  # pylint:disable=no-member
+            op_class = getattr(qml.ops.qubit, op_name)  # pylint:disable=no-member
+            if op_class.has_generator:
+                if op_class.num_wires == AnyWires:
+                    has_generator_types_anywires.append(op_class)
+                elif op_class.num_wires == base.num_wires:
+                    has_generator_types.append(op_class)
+        # Ensure op_class.num_wires == base.num_wires before op_class.num_wires == AnyWires
+        has_generator_types += has_generator_types_anywires
```

Comment:
It looks like `has_generator_types_anywires` is not being used past this line. What's the purpose of defining this variable?

Reply:
I defined this list separately and then appended to `has_generator_types` to ensure that the more general operators with `AnyWires` are checked after the more specific ones. In the previous implementation, this was achieved by using an order specified by hand; here, as the list is constructed at runtime, this ensures these operations are checked last. It's just because to construct this list with only one pass of the loop, I have to store both cases in parallel. I could remove this if I loop through just extracting those without `AnyWires`, and then a second time for the `AnyWires` cases, but this would be slower?

Reply:
Ah I see, I missed the comment in the line above. No that should be fine.